addon/components/frost-link.js

```diff
@@ -309,6 +325,18 @@ export default LinkComponent.extend(PropTypeMixin, HookMixin, SpreadMixin, {
   },
 
   /**
+   * Open a link for an external location.
+   */
+  _openExternalLink () {
+    const href = this.get('href')
+    // If we want to open a new tab, the target will be set accordingly when we setup the routing.
+    // If we want to stay in the same tab, we will set the href.
+    if (this._shouldOpenInSameTab()) {
+      window.location.href = href
```

Comment:
Currently the implementation of the link is the following:
primary link: new tab
secondary link: same tab

I followed the same concept here.  I don't think there is any way to open the link in the same tab if it's a primary link.